meta-resin-common/recipes-devtools/kernel-modules-headers/kernel-modules-headers.bb

```diff
@@ -36,13 +36,15 @@ do_compile() {
     ${S}/gen_mod_headers ./kernel_modules_headers ${STAGING_KERNEL_DIR} ${DEPLOY_DIR_IMAGE} ${ARCH} ${TARGET_PREFIX} "${CC}" "${HOSTCC}"
 
     # Sanity test
-    test_arch=$(find kernel_modules_headers/  | xargs file | grep ELF | xargs -I a bash -c 'if ! echo "a" | grep -Fiq "${ARCH}" ; then echo "Did not find ${ARCH}"; fi')
+    test_arch=$(find kernel_modules_headers/  | xargs file | grep ELF | xargs -I {} bash -c 'if ! echo "{}" | grep -Fiq "${ARCH}" ; then echo "Did not find ${ARCH}"; fi')
     if [ ! -z "$test_arch" ]; then
         bberror "Wrong arch found in ELF files"
+        exit 1
     fi
-    test_interpreter=$(find kernel_modules_headers/  | xargs file | grep ELF | xargs -I a bash -c 'if echo "a" | grep -Fiq "sysroot" ; then echo "Found sysroot in interpreter" ; fi')
+    test_interpreter=$(find kernel_modules_headers/  | xargs file | grep ELF | xargs -I {} bash -c 'if echo "{}" | grep -Fiq "sysroot" ; then echo "Found sysroot in interpreter" ; fi')
     if [ ! -z "$test_interpreter" ]; then
         bberror "Sysroot keyword found in interpreter ELF files"
```

Comment:
Use bbfatal.

Reply:
@agherzan thats what I thought.

but the bash version of `bbfatal` points to `bberror`.

```
../poky/bitbake/lib/bb/build.py-414-                elif cmd == 'bberror':
../poky/bitbake/lib/bb/build.py-415-                    bb.error(value)
../poky/bitbake/lib/bb/build.py:416:                elif cmd == 'bbfatal':
../poky/bitbake/lib/bb/build.py-417-                    # The caller will call exit themselves, so bb.error() is
../poky/bitbake/lib/bb/build.py-418-                    # what we want here rather than bb.fatal()
../poky/bitbake/lib/bb/build.py-419-                    bb.error(value)
```

Reply:
Interesting. OK.